spec/datetimeformat.html

```diff
@@ -803,10 +803,10 @@ <h1>DateTimeStyleFormat ( _dateStyle_, _timeStyle_, _styles_ )</h1>
           1. If _dateTimeRangeFormat_ has a [[rangePatterns12]] field, then
             1. Set _format_.[[rangePatterns12]] to _dateTimeRangeFormat_.[[rangePatterns12]].
           1. Return _format_.
-        1. If _timeStyle_ is not *undefined*, then
-          1. Return _timeFormat_.
-        1. Assert: _dateStyle_ is not *undefined*.
-        1. Return _dateFormat_.
+        1. If _timeStyle_ is *undefined*, then
+          1. Assert: _dateStyle_ is not *undefined*.
+          1. Return _dateFormat_.
+        1. Return _timeFormat_.
```

Comment:
I think I just disagree with #458 here. The "fallbacks" pattern more ergonomically translates to comprehensible code[^1] IMO.
```suggestion
        1. Else if _dateStyle_ is not *undefined*, then
          1. Return _dateFormat_.
        1. Else,
          1. Assert: _timeStyle_ is not *undefined*.
          1. Return _timeFormat_.
```

[^1]: e.g.,
      ```js
      if (dateStyle !== undefined && timeStyle !== undefined) {
        …
      } else if (dateStyle !== undefined) {
        return dateFormat;
      } else {
        assert(timeStyle !== undefined);
        return timeFormat;
      }
      ```